### DISCUSSION TITLE
Could you use slots for showing authorized or unauthorized site?

### DISCUSSION THREAD

**Discussion by maartensson** [leptos-rs/leptos#2025] [Q&A] [answered]
I am just looking through all the examples and I saw the one with slots. I cant read anything about it in the docs and it does not say much about it in the readme. 

It makes sense that you can use it for anything conditional, but I am wondering if showing things based on authstate is a good use case?

**Comment by gbj** [comment, marked as answer]
Just to be clear, here are the docs for [the `#[slot]` macro](https://docs.rs/leptos/latest/leptos/attr.slot.html). It is used to allow you to provide some kind of typed as a prop to another component, and to build that data using the component syntax itself. 

It doesn't have anything to do with conditionality in itself, that is just the example given. It is mostly a syntax sugar: in the example you're referring to, writing 
```
        <SlotIf cond=is_even>
            // The slot name can be emitted if it would match the slot struct name (in snake case).
            <Then slot>"even"</Then>
            // Props are passed just like on normal components.
            <ElseIf slot cond=is_div5>"divisible by 5"</ElseIf>
            <ElseIf slot cond=is_div7>"divisible by 7"</ElseIf>
            <Fallback slot>"odd"</Fallback>
        </SlotIf>
```
is an alternative to something like
```
        <SlotIf cond=is_even
          then="even"
          else_if=vec![
            ElseIf {
              cond=is_div5,
              children="divisible by 5"
            },
            ElseIf {
              cond=is_div7,
              children="divisible by 7"
            }
          ],
          fallback="odd"
        >
```
